src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -131,6 +141,13 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken)
             // This is the earliest opportunity to fetch satellite assemblies for this selection.
             await CultureProvider.LoadCurrentCultureResourcesAsync();
 
+            var manager = Services.GetRequiredService<ComponentApplicationLifetime>();
```

Comment:
How would it look if we wanted to make this feature linkable-out? Instead of baking it directly into `WebAssemblyHost`, could we make the initialization something the developer calls themselves from their `Program.cs` logic? So if they didn't have that call, there wouldn't be any references to all these types and the whole thing would disappear.

I know we could argue that this specific feature isn't going to single-handedly make a giant difference, but we do need to be in the habit of designing new features for linkability so that (1) it doesn't grow too fast as we add features, and (2) we don't have to work so hard to reason about how each extra system API we reference while maintaining existing code will balloon the output for everyone.

I know this would be a disruptive change right now. We have a bit more time before the preview 2 deadline so it probably is possible to look into in the short term, but I'm OK with tracking it as a follow-up if you prefer.

Reply:
I'm glad that you bring this up.

Initializations like these have been building up across our hosts (prerender, wasm, and server). I have a design in mind to "externalize" them and make them a general concept that third-party libraries can also leverage. 

I'm not going to deal with this in this PR, but I'll chat with you about it in the following weeks.

Reply:
I filed an issue so that we don't loose track of this, but this is what I'm thinking about https://github.com/dotnet/aspnetcore/issues/30304